### PR TITLE
Document the new/upcoming issue labels

### PR DESCRIPTION
**Pull request by sauclovian-g**
Update the mergify config with the new issue label strings.

Drop the keep-branch-on-merge feature; have it just always remove pull request branches at merge time. (It's not exactly difficult to put them back if that's what you want, and it appears the one time this feature's been used the branch hung around uselessly for years.)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -57,6 +58,338 @@ library](https://github.com/aws/s2n-tls), continue to succeed. The
...
+change the type label.
+
+These labels have a common text prefix so they sort together.
+Issues should ordinarily have exactly one type label.
+
+### Issue state (yellow)
+
+We are currently not using issue state labels.
+
+### Technical topics (pale green)
```
Comment (mccleeary-galois) [inline CONTRIBUTING.md:247]: I thought we were going to add `missing-cryptol-features` as well?

**Review by mccleeary-galois — APPROVED** [review]
I am happy with these labels overall, I think it would be good to give others a chance to review as well.

**Comment by sauclovian-g** [comment]
If there's anything else I've forgotten, this is a good time to notice :-)

(Also while I'm posting maybe I should note that I changed the spelling of the category prefixes from ```foo/bar``` to ```foo: bar``` because apparently you can do that and it looks a lot nicer.)

**Review by RyanGlScott — COMMENTED** [review]
I was involved with initial discussions on what this label documentation should look like, so naturally I am happy with the state it's ended up in here.

One thing that perhaps is worth calling out: in the past, folks have generally added new labels on an as-needed basis. Now that we are going through the effort of imposing a particular set of processes on labels (and documenting each one), if we want to add a label in the future, it might be better for the label-adder to submit a PR first to amend this document _before_ adding the label to GitHub itself. Do you think this warrants a mention here in this documentation?

**Review by andrew-bivin — APPROVED** [review]

**Comment by sauclovian-g** [comment]
Yes, good point.

**Pull request merged**